docker/witnet-rust/wip0010-recovery.sh

```diff
@@ -0,0 +1,153 @@
+
+# Just a pretty logging helper
+function log {
+  echo "[WIP0010_RECOVERY] $1"
+}
+
+# A helper for calculating ETAs
+function eta {
+  START=$1
+  NOW=$2
+  PROGRESS=$3
+  if [ "$PROGRESS" == "00" ]; then
+    echo "will be shown as synchronization moves forward..."
+  else
+    ELAPSED=$(( NOW - START ))
+    SPEED=$((PROGRESS * 1000 / ELAPSED))
+    REMAINING_PROGRESS=$(( 10000 - PROGRESS ))
+    REMAINING_TIME=$((REMAINING_PROGRESS / (SPEED / 1000) ))
+    echo $(( REMAINING_TIME / 60 )) minutes $((REMAINING_TIME % 60)) seconds
+  fi
+}
+
+# This script can be skipped by setting environment variable SKIP_WIP0010_RECOVERY to "true"
+if [[ "$SKIP_WIP0010_RECOVERY" == "true" ]]; then
+  log "Skipping WIP-0010 recovery"
+  exit 0
+fi
+
+# Make sure the arguments make sense
+if ! command -v "$WITNET_BINARY" &> /dev/null; then
+  log "ERROR: The provided witnet binary (first argument to this script) is not a valid executable file: $WITNET_BINARY"
+  exit 1
+fi
+if [ ! -f "$WITNET_CONFIG_FILE" ]; then
+  log "ERROR: The provided witnet configuration file (second argument to this script) is not a valid configuration file: $WITNET_CONFIG_FILE"
+  exit 2
+fi
+
+# Read configuration (e.g. node server address) from config file
+log "Using configuration file at $WITNET_CONFIG_FILE"
+HOST=$(grep "server_addr" "$WITNET_CONFIG_FILE" | sed -En "s/server_addr = \"(.*)\"/\1/p" | sed -En "s/0.0.0.0/127.0.0.1/p" )
+ADDRESS=$(echo "$HOST" | cut -d':' -f1)
+PORT=$(echo "$HOST" | cut -d':' -f2)
+
+# Check connection to local witnet node
+TIME_TO_NEXT_RETRY=5
+log "Checking connection to local witnet node at $HOST"
+while true
+  if nc -zv "$ADDRESS" "$PORT" &>/dev/null; then
+    log "Successful connection to local witnet node at $HOST"
+    break
+  else
+    log "ERROR: Failed to connect to local witnet node at $HOST"
+    log "Retrying in $TIME_TO_NEXT_RETRY seconds"
+    sleep "$TIME_TO_NEXT_RETRY"
+    TIME_TO_NEXT_RETRY=$(( 2 * TIME_TO_NEXT_RETRY ))
+  fi
+do true; done
+
+# Check whether the local witnet node is below WIP-0010 "common checkpoint" (#248839)
+if [[ "$($WITNET_BINARY node blockchain --epoch 248839 --limit 1 2>&1 | wc -l)" == "5" ]]; then
```

Comment:
This seems a bit hacky, wouldn't it be better to grep for `block for epoch`?

Output if there is a block above that checkpoint:

```
Loading config from: witnet.toml
Setting log level to: ERROR, source: Env
block for epoch #248839 had digest 47eb3069ad98b81c89bd0691f89885387d51505ed3f45ba2b0a240582bb75b0c
```

Output if there are no blocks above that checkpoint:
```
Loading config from: witnet.toml
Setting log level to: ERROR, source: Env
```